pkg/ui/react-app/src/components/ListTree.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+import { InputProps, Collapse, ListGroupItem, ListGroup } from 'reactstrap';
+
+export interface NodeTree {
+  name: string;
+  children?: NodeTree[];
+}
+
+interface NodeProps extends InputProps {
+  node: NodeTree | null;
+}
+
+const ListTree: React.FC<NodeProps> = ({ id, node }) => {
+  type mapping = {
+    [key: string]: boolean;
+  };
+
+  const [state, setState] = useState<mapping>({});
+  const toggle = (e: React.MouseEvent<HTMLDivElement>) => {
+    const el = e.target as HTMLDivElement;
+    const id = el.getAttribute('id');
+    if (id) {
+      setState({ ...state, [id]: !state[id] });
+    }
+  };
+
+  const mapper = (nodes: NodeTree[], parentId?: any, lvl?: any) => {
+    return nodes.map((node: NodeTree, index: number) => {
+      const id = `${index}-${parentId ? parentId : 'top'}`.replace(/[^a-zA-Z0-9-_]/g, '');
+      const item = (
+        <React.Fragment>
+          <ListGroupItem
+            className={`bg-transparent p-0 border-0 ${parentId ? `rounded-0 ${lvl ? 'border-bottom-0' : ''}` : ''}`}
+          >
+            {
+              <div className={`d-flex align-items-center`} style={{ paddingLeft: `${25 * lvl}px` }}>
+                {node.children && (
+                  <div className="pl-0 btn text-primary" style={{ cursor: 'inherit' }} color="link">
+                    {state[id] ? '-' : '+'}
```

Comment:
This is already really good! I wonder tho if we can use some unicode symbol instead to better represent this?